openlibrary/catalog/marc/fast_parse.py

```diff
@@ -17,7 +17,9 @@
 def translate(bytes_in, leader_says_marc8=False):
     """
     Converts MARC8 to unicode
+    :param bytes_in bytes:
     """
+    assert isinstance(bytes_in, bytes)
```

Comment:
See #3717 for an alternative approach.

Reply:
The `except AttributeError` in PR #3717 should never trigger, the function should always take `bytes`, of either MARC8 or UTF-8 encoded data, and return a unicode string. I've just updated the docstring to make that clearer. This is a deprecated function, and not used in the import path, but is still used for the online view interface.  As part of clearing up what format the Binary MARC class should be, it becomes clear that the input should always be `bytes`.   If there are tests sending `str`, they are doing the wrong thing. I think I have caught all the ambiguous cases and converted them to `bytes`.

Reply:
@cclauss  This `assert` statement could be removed, I left it there more as documentation to show `bytes_in` is expected to be `bytes`, but the docstring should cover that. I can remove line 22 if you think it's unnecessary.

Reply:
OK. Understood.  Let’s leave the assert in place.  It is better to be strict about expected types.